File: android/src/main/kotlin/com/atishoo/dingtalk/dingtalk_sdk/constant/Response.kt
package com.atishoo.dingtalk.dingtalk_sdk.constant

class AuthResponse {
    var authCode:String = ""
    var error:String = ""
    var state:String = ""

    constructor(authCode: String, error: String, state: String) {
        this.authCode = authCode
        this.error = error
        this.state = state
    }
}

File: lib/response.dart
class BaseResponse {
  factory BaseResponse.create(String method, Map args) {
    if (method == 'onAuthResp') {
      return AuthResponse(args['authCode'], args['error'], args['state']);
    } else {
      return BaseResponse();
    }
  }

  BaseResponse();
}

class AuthResponse extends BaseResponse {
  final String authCode;
  final String error;
  final String state;

  AuthResponse(this.authCode, this.error, this.state);
}
